lib/utils.js

```diff
@@ -33,7 +34,12 @@ function getRefType (ref) {
     if (validUrl.isWebUri(val)) {
       return 'web'
     }
-    if ((validUrl.isUri(val) && val.indexOf('file:') === '0') || (val.indexOf('.json') > 0) || isValidPath(val)) {
+    if (
+      (validUrl.isUri(val) && val.startsWith('file:')) ||
+      val.includes('.json') ||
+      val.match(yamlSuffixRegex) ||
+      isValidPath(val.split['#'])
```

Comment:
BTW, `val.split['#']` would always return `undefined`.
what's the purpose of `isValidPath(val.split['#'])`?

Reply:
Care to explain this guys?